### PR TITLE
docs: Restructure getting started guide

### LINKED ISSUE #1826: Add unit tests to getting started guide
The unit tests are something we want to drive users towards while developing Vector configuration files, especially for error-prone transforms like `regex_parser`, `lua`, etc. I'd like to add simple unit tests into the getting started guide and remove the manual testing parts until the very end. Actually running Vector is the last thing a user should do, just to smoke test and see it all working.

### PR DESCRIPTION
**Pull request by Jeffail**
This PR restructures the getting started guide in order to showcase our dapper new unit testing functionality.

Closes #1826

### CODE REVIEW

File: website/docs/setup/guides/getting-started.md [hunk 1]
```diff
@@ -106,22 +100,101 @@ order to print the full event structure:
...
+sure, right?
+
+## 4. Test it
+
+No one is saying that explosions aren't cool, but you should be doing that in
+your own time and with your own infrastructure.
+
+Vector takes testing seriously and has support for writing
+[unit tests][docs.guides.unit_testing] as part of your config, so let's humor it
+by writing one for our log parser:
```
Comment (binarylogic) [inline website/docs/setup/guides/getting-started.md]: I think the drive here is that testing _should_ be easier. Manual testing would require a local ES instance or actually deploying Vector to production. So I'd rephrase this like:

So how do we know all of this works? We could setup a local Elasticsearch instance, or deploy Vector to production and hope for the best. Vector is different! With Vector you can write simple unit tests directly in your configuration file, just like you would with code. Now you know your Vector config file works without all of the time wasting manual testing. Here's how:

File: website/docs/setup/guides/getting-started.md [hunk 2]
```diff
@@ -130,62 +203,201 @@ order to print the full event structure:
...
+
+test 'test apache regex':
+
+check transform 'apache_parser' payloads (events encoded as JSON):
+  input: {"timestamp":"2020-02-19T14:15:45.997803Z","message":"172.128.80.109 - Bins5273 656 [2019-05-03T13:11:48-04:00] \"PUT /mesh\" 406 10272"}
+  output: {"host":"172.128.80.109","user":"Bins5273","bytes_out":"10272","timestamp":"2019-05-03T13:11:48-04:00","path":"/mesh","bytes_in":"656","method":"PUT","status":"406"}
 ```
 
-Oh snap! This time we get something like:
+This is called an _inspection_ test and is super cool. It allows us to
```
Comment (binarylogic) [inline website/docs/setup/guides/getting-started.md]: Is a flag missing above that would make this an "inspection" test? Or is the vanilla `vector test` command doing this?

Comment (binarylogic) [reply website/docs/setup/guides/getting-started.md]: I see now. Hm. This section feels somewhat long. I'm wondering if we could shorten this step by adding the conditions right off the bat? Maybe add 1 or 2 so it's not overwhelming?

File: website/docs/setup/guides/getting-started.md (hunk 1 above)
Comment (Jeffail) [reply website/docs/setup/guides/getting-started.md]: Agreed, reworded this section, how do you feel about the new version?

File: website/docs/setup/guides/getting-started.md (hunk 2 above)
Comment (Jeffail) [reply website/docs/setup/guides/getting-started.md]: I've reworded it to make it clearer. Agree it's long, I'll try and cut it down some more tomorrow. I think it's important to demo the inspections first since that's what we're using instead of manual testing (otherwise we don't see the events).

The alternative is to add a flag `vector test --inspect` that forces all test outputs to be inspections even if they have conditions that pass. Then we can add the full unit test in one step.

Comment (binarylogic) [reply website/docs/setup/guides/getting-started.md]: I'd be in favor of an `--inspect` flag or demonstrating a simple failure. But I would like to shorten this. A getting started guide should be simple and quick to complete.

Comment (Jeffail) [reply website/docs/setup/guides/getting-started.md]: I'll make the regex fail in the guide, that'll allow us to showcase everything with just one test stage.

**Comment by Hoverbear** [comment]
Good job. :)
